node/node.go

```diff
@@ -114,6 +116,47 @@ func NewNode(reg *prometheus.Registry, config *Config, pubIPProvider pubip.Provi
 	// Create ChainState Client
 	cst := eth.NewChainState(tx, client)
 
+	var keyPair *core.KeyPair
+	var blsClient blssignerV1.SignerClient
+	if config.PrivateBls != "" {
+		nodeLogger.Info("using local keystore private key for BLS signing")
+		// Generate BLS keys
+		keyPair, err = core.MakeKeyPairFromString(config.PrivateBls)
+		if err != nil {
+			return nil, err
+		}
+
+		config.ID = keyPair.GetPubKeyG1().GetOperatorID()
+	} else {
+		pkBytes, err := hex.DecodeString(config.BLSPublicKeyHex)
```

Comment:
personally find this a bit confusing to read.
Is the intention that:
- test mode: bls private key
- prod mode: bls sk file OR bls remote signer
?

in config.go above you first check if !testmode, then build the bls remote signer, so I think here you should also duplicate the logic (for redundancy and future buggy refactor proofing) - check
```
if !testMode && config.PrivateBls != ""
```
or something. Haven't thought this through, but I personally don't like any implicit stuff (like implicitly turning bls remote signer when some other flag is present. Prefer everything to have an "enabled" switch. Perhaps you can have a `SignerType` flag, and then crash when that thing is set to private-key and testMode is set to disabled?

Reply:
> test mode: bls private key
> prod mode: bls sk file OR bls remote signer
> ?

Yes - I am not aware where test mode is use? maybe I think in inabox testing but yes that's the intention - in prod you can use either encrypted sk file or remote signer

Reply:
so what I can do is - have an explicit flag which switches between sk file and bls remote signer. `PrivateBls` is not really used in prod. maybe @ian-shim can confirm but I think thats true.

Reply:
maybe we should eventually remove out test node and use either sk file or remote signer as that clearly mimics prod usage

Reply:
the ugly thing about testMode and prod mode with sk is - both these cases populate `privateBls` and that's why checking testMode is not going to work here. since `privateBls` is populated in non testMode too. https://github.com/Layr-Labs/eigenda/pull/745/files#diff-923e406f4251573464f7748834b59f4747b3f6ec4eadda667614e2875a3b811aR166-R174

I think ideal would be to remove this testMode and eventually put the prod signing behind eigensdk as I said in above comment.

Reply:
Test mode I think is just used to allow passing a sk directly? Think it should be allowed to simplify tests.

Reply:
if that would remain - in current way - `privateBls` will be populated for both. I guess we could change eventually when we abstract out signing behind eigensdk